common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2968,8 +2975,12 @@ def _copy_subdag(self, user_id, destination_version, block_key, source_blocks, d
                 if getattr(destination_block.edit_info, key) is None:
                     setattr(destination_block.edit_info, key, val)
 
-        # introduce new edit info field for tracing where copied/published blocks came
-        destination_block.edit_info.source_version = new_block.edit_info.update_version
+        # After revert_to_published(), the source_version of both draft and published blocks are same.
+        # Publishing the block after that point replaces the published block's edit info with the draft
+        # block's edit info - making has_changes() True even when no changes have been made.
+        # Prevents the block from getting stuck in the draft state.
+        # pylint: disable=line-too-long
```

Comment:
"If the block we are copying from was itself a copy, then just reference the original source, rather than the copy."